plugins/scaffolder-backend/src/index.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { TemplaterConfig, createTemplater } from './lib/templater';
+import { createStorage, StorageConfig } from './lib/storage';
+import express from 'express';
+
+export const createScaffolder = (config?: TemplaterConfig & StorageConfig) => {
+  const store = createStorage(config);
+
+  const templater = createTemplater(config);
+
+  const router = express.Router();
+
+  router.get('/v1/templates', async (_, res) => {
+    const templates = await store.list();
+    res.status(200).json(templates);
+  });
+
+  router.post('/v1/job/create', async (_, res) => {
```

Comment:
+1 for job creation and tracking. What do you think about scaling this to multiple instances? Thinking mostly about API design and if there's some way to keep it simple.

Reply:
POST job description json to /v1/jobs -> 201 Created + Location: /v1/jobs/183636 + json payload with job description incl generated id

Then poll /v1/jobs/183636 repeatedly -> json payload with job description + runtime status

Reply:
Was thinking about how that would work with horizontal scaling of the backend. Obviously a DB would solve it, but it might be unfortunate if it's required, although maybe that's just something every deployment is gonna need.

Reply:
Yeah I just meant that that api makes sense either way and scales to adding a db later.

Yeah we should at least make it super simple to add a shared db

Reply:
Yeah. It'll depend on the deployment model as well - check #686. There, you may well give an env var if it's launched standalone, but if you instantiate the router you would just give it a connection pool directly

Reply:
Yeah I've used SQLite/PostgreSQL combo before and imo it works pretty nicely. When mixing databases you usually might end up using some feature that's not supported in the other. Since SQLite's feature set is more limited you end up avoiding more arcane feature of PostgreSQL. Still need to stick to the correct syntax and ofc test towards both, but :itssomething:

Reply:
Yeah. It'll depend on the deployment model as well - check #686. There, you may well give an env var if it's launched standalone, but if you instantiate the router you would just give it a connection pool directly

Reply:
Defo gonna talk bout settling on redis soon!

Reply:
I'm +1 for redis.

Massive fan sat here.

Reply:
+1 on Redis